src/WorldStorage/WSSAnvil.cpp

```diff
@@ -1472,17 +1472,33 @@ cBlockEntity * cWSSAnvil::LoadSignFromNBT(const cParsedNBT & a_NBT, int a_TagIdx
 
 
 
+bool cWSSAnvil::CheckEntityType(const char * a_IDTag, size_t a_IDTagLength, const AStringVector & a_ExpectedTypes)
+{
+	for (auto & et : a_ExpectedTypes)
+	{
+		if (strncmp(a_IDTag, et.c_str(), a_IDTagLength) == 0)
+		{
+			return true;
+		}
+	}
+	return false;
+}
+
+
+
+
+
 void cWSSAnvil::LoadEntityFromNBT(cEntityList & a_Entities, const cParsedNBT & a_NBT, int a_EntityTagIdx, const char * a_IDTag, size_t a_IDTagLength)
 {
-	if (strncmp(a_IDTag, "Boat", a_IDTagLength) == 0)
+	if (CheckEntityType(a_IDTag, a_IDTagLength, { "Boat", "minecraft:boat" }))
```

Comment:
These will probably have a poor performance, because you're creating the AStringVector each time when loading an entity. Consider pulling these out to a separate static const variable:
```cpp
  static const AStringVector entityTypesBoat({"Boat", "minecraft:boat"});
  // ...
  if (CheckEntityType(a_IDTag, a_IDTagLength, entityTypesBoat))
```